### PR TITLE
[CHI-73] Add drawer component

### PR DESCRIPTION
**Pull request by dani-cl-madrid**
Drawer component has been added.
Backdrop has been unified for modal, spinner and drawer (Not included -active support).
Not included any JavaScript.

### CODE REVIEW

File: src/chi/components/drawer/drawer.scss [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+    & {
+      background-color: set-color(white);
+      box-shadow: 0 0 25px 10px rgba(set-color(black), 0.15);
+      display: none;
+      flex-direction: column;
+      overflow-x: hidden;
+      overflow-y: auto;
+      pointer-events: auto;
+      position: absolute;
```
[the remark this answers is not in the document]
Comment (dani-cl-madrid) [reply src/chi/components/drawer/drawer.scss]: You were completely right. 😓

File: src/chi/components/backdrop/backdrop.scss [hunk 2]
```diff
@@ -0,0 +1,21 @@
+@import 'mixins';
+
+.a-backdrop {
```
[the remark this answers is not in the document]
Comment (dani-cl-madrid) [reply src/chi/components/backdrop/backdrop.scss]: I have defined a set of z-index values to be used in the whole project. Very related to CHI-131 Add z-index utility.

File: src/website/views/components/drawer.pug [hunk 3]
```diff
@@ -0,0 +1,212 @@
+---
+title: Drawer
+---
+p.-text
+  | Drawer primary use case is to slide in navigation on small devices. However, it can be also utilized on
+  | standard desktop displays.
+
+h3 Examples
+
+p Drawers support sliding in from top, right, bottom, or left. They
```
Comment (mattnickles) [inline src/website/views/components/drawer.pug]: Either a sentence is missing here or we just need to remove "They"

File: src/website/assets/styles/_docs-header.scss [hunk 4]
```diff
@@ -7,6 +7,7 @@
 
     @include respond-to(lg) {
       position: fixed;
+      z-index: 2006;
```
Comment (mattnickles) [inline src/website/assets/styles/_docs-header.scss]: Could be problematic for non-mobile scenarios as it hides the drawers close button. Might be safer to remove for now.

**Pull request merged**